app/helpers/elasticsearch_helper.rb

```diff
@@ -32,11 +33,31 @@ def taxon_search(prefix, tax_levels = TaxonCount::NAME_2_LEVEL.keys)
         }
       end
     end
-    matching_taxa.values
+
+    if project_id.present?
+      filter_by_project(matching_taxa.values, project_id)
+    else
+      matching_taxa.values
+    end
   end
 
   private
 
+  # Took 250ms in local testing on real data
+  def filter_by_project(matching_taxa, project_id)
+    project_tax_ids = Set.new(TaxonCount
+      .joins(pipeline_run: { sample: :project })
+      .where(samples: { project_id: project_id })
+      .where(tax_id: [matching_taxa.map { |taxa| taxa["taxid"] }])
+      .where("tax_id > 0") # negative numbers mean unknown... see TaxonLineage
+      .where(count_type: ["NT", "NR"])
+      .where("count > #{ReportHelper::MINIMUM_READ_THRESHOLD}")
```

Comment:
prefer `.where("count > ?", ReportHelper::MINIMUM_READ_THRESHOLD)` as a general practice :D

Reply:
for security reasons